### PR TITLE
qcom: make camera_process_frame more readable

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

**Comment by deanlee** [comment]
going to split camera_process_frame into smaller functions

**Comment by ZwX1616** [comment]
looks good so far. Is it ready to merge?

**Comment by deanlee** [comment]
Not ready yet, I'll do more checks after rebase master

File: selfdrive/camerad/cameras/camera_qcom.cc [hunk 1]
```diff
@@ -2055,96 +2045,111 @@ static void* ops_thread(void* arg) {
...
+static void update_lapmap(MultiCameraState *s, const CameraBuf *b, const int cnt) {
+  const size_t width = b->rgb_width / NUM_SEGMENTS_X;
+  const size_t height = b->rgb_height / NUM_SEGMENTS_Y;
+  static std::unique_ptr<uint8_t[]> rgb_roi_buf = std::make_unique<uint8_t[]>(width * height * 3);
+  static std::unique_ptr<int16_t[]> conv_result = std::make_unique<int16_t[]>(width * height);
 
   // sharpness scores
   const int roi_id = cnt % ARRAYSIZE(s->lapres);  // rolling roi
-  int roi_x_offset = roi_id % (ROI_X_MAX-ROI_X_MIN+1);
-  int roi_y_offset = roi_id / (ROI_X_MAX-ROI_X_MIN+1);
```
Comment (deanlee) [inline selfdrive/camerad/cameras/camera_qcom.cc:2069]: It looks like this is a previous bug.it should be` roi_id / (ROI_X_MAX-ROI_X_MIN+1);`

Comment (ZwX1616) [reply selfdrive/camerad/cameras/camera_qcom.cc:2069]: it should be correct. col(x) is the mod and row(y) is the integer division of stride, no?

Comment (deanlee) [reply selfdrive/camerad/cameras/camera_qcom.cc:2069]: my bad.reverted

**Pull request merged**